pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/plan/PhysicalPlanVisitor.java

```diff
@@ -105,8 +110,24 @@ public MultiStageOperator visitWindow(WindowNode node, PlanRequestContext contex
 
   @Override
   public MultiStageOperator visitSetOp(SetOpNode setOpNode, PlanRequestContext context) {
-    throw new UnsupportedOperationException(
-        "Stage node of type SetOpNode: " + setOpNode.getSetOpType() + " is not supported!");
+    List<MultiStageOperator> inputs = new ArrayList<>();
+    for (StageNode input : setOpNode.getInputs()) {
+      MultiStageOperator visited = input.visit(this, context);
+      inputs.add(visited);
+    }
+    switch (setOpNode.getSetOpType()) {
+      case UNION:
+        return new UnionOperator(context.getOpChainExecutionContext(), inputs,
+            setOpNode.getInputs().get(0).getDataSchema());
```

Comment:
this and others below: seems like it is assuming it always gets Union "all=true"?
do we plan to support all=false going forward? if so can we add check on `setOp.all` ? and also add a ignored test in SetOp.json test cases?

Reply:
For now UNION is parsed to UNION ALL + DISTINCT, so for UNION, all is always true.
For INTERSECT and MINUS, the all is always false.

We can support it the check for all later if parser side can support it.